state-surgery/state/state_test.go

```diff
@@ -0,0 +1,35 @@
+package state_test
+
+import (
+	"encoding/json"
+	"os"
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/state-surgery/solc"
+	"github.com/ethereum-optimism/optimism/state-surgery/state"
+	"github.com/stretchr/testify/require"
+)
+
+var layout solc.StorageLayout
+
+func init() {
+	data, err := os.ReadFile("./testdata/layout.json")
+	if err != nil {
+		panic("layout.json not found")
+
+	}
+	if err := json.Unmarshal(data, &layout); err != nil {
+		panic("cannot unmarshal storage layout")
+	}
+}
+
+func TestComputeStorageSlots(t *testing.T) {
+	values := state.StorageValues{}
+	values["time"] = 12
+	values["addr"] = "0xEA674fdDe714fd979de3EdF0F56AA9716B898ec8"
+	values["boolean"] = true
+
+	slots, err := state.ComputeStorageSlots(&layout, values)
```

Comment:
What is this test asserting? It looks like it only asserts that the encoding was successful, and doesn't check the values of the storage slots themselves.

Reply:
I can add assertions for this, was planning on thorough testing in a follow up PR because this is critical code. I have a few ideas on ways to test this thoroughly, it still needs a bit of functionality to be ready for our usage

Reply:
I'm fine adding testing in a separate PR if the tests are substantial and will need dedicated review.